posthog/batch_exports/service.py

```diff
@@ -100,14 +129,47 @@ async def pause_schedule(temporal: Client, schedule_id: str, note: str | None =
     await handle.pause(note=note)
 
 
-def unpause_batch_export(temporal: Client, batch_export_id: str, note: str | None = None) -> None:
+def unpause_batch_export(
+    temporal: Client, batch_export_id: str, note: str | None = None, backfill: bool = False
+) -> None:
     """Pause this BatchExport.
 
-    We pass the call to the underlying BatchExportSchedule. This exists here as a convinience so that users only
-    need to interact with a BatchExport.
+    We pass the call to the underlying Temporal Schedule. Additionally, we can trigger a backfill
+    to backfill runs missed while paused.
+
+    Args:
+        temporal: The Temporal client to execute calls.
+        batch_export_id: The ID of the BatchExport to unpause.
+        note: An optional note to include in the Schedule when unpausing.
+        backfill: If True, a backfill will be triggered since the BatchExport's last_paused_at.
+
+    Raises:
+        BatchExportIdError: If the provided batch_export_id doesn't point to an existing BatchExport.
     """
-    BatchExport.objects.filter(id=batch_export_id).update(paused=False)
-    unpause_schedule(temporal, schedule_id=batch_export_id, note=note)
+    batch_export = BatchExport.objects.get(id=batch_export_id)
```

Comment:
I think this will raise if the export doesn't exist rather than return something falsy

Reply:
Yeah, get raises a `DoesNotExist` exception. This means we don't need to check for `None`, and DRF will (I think) just return a 404.

Reply:
Ah, nevermind, there is a `get_object_or_404` precisely for this.